### PR TITLE
Add HBASE-25292 and HBASE-25336 to HBase 2.4

### PR DESCRIPTION
**Pull request by lfrancke**
# Description

This PR back-ports two fixes from HBase `2.5` to `2.4.12` and `2.4.17`.

Previously regions were stuck in transitioning when the regionservers were restarted e.g. by using `k rollout restart sts prod1-hbase-regionserver-default`.

These fixes are:
[HBASE-25292](https://issues.apache.org/jira/browse/HBASE-25292): Improve InetSocketAddress usage discipline
This patch fixed the regions stuck in transitioning.

[HBASE-25336](https://issues.apache.org/jira/browse/HBASE-25336): Use Address instead of InetSocketAddress in RpcClient implementation
This patch probably only effects hbase clients - but is needed as we offer HBase restservers.

Additionally we made sure that the two HBase images work against `23.7.0` and `0.0.0-dev` Stackable operators.

## Definition of Done Checklist

- Not all of these items are applicable to all PRs, the author should update this template to only leave the boxes in that are relevant
- Please make sure all these things are done and tick the boxes
 
```[tasklist]
- [x] Changes are OpenShift compatible
- [x] All added packages (via microdnf or otherwise) have a comment on why they are added
- [x] Things not downloaded from Red Hat repositories should be mirrored in the Stackable repository and downloaded from there
- [x] All packages should have (if available) signatures/hashes verified
- [x] Does your change affect an SBOM? Make sure to update all SBOMs
```

### CODE REVIEW

File: hbase/Dockerfile [hunk 1]
```diff
@@ -159,5 +159,10 @@ ENV HBASE_CONF_DIR=/stackable/hbase/conf
 ENV HOME=/stackable
 ENV PATH="${PATH}":/stackable/bin:/stackable/hbase/bin
 
+
+USER 0
+RUN ln -s /stackable/jmx/regionserver.yaml /stackable/jmx/region-server.yaml
```
Comment (maltesander) [inline hbase/Dockerfile:164]: Changed in https://github.com/stackabletech/docker-images/pull/494 and adapted in operator https://github.com/stackabletech/hbase-operator/blame/6f9b4d0b50f72663be2568aea8979f2b86dc8558/rust/crd/src/lib.rs#L360
```suggestion
RUN ln -s /stackable/jmx/regionserver.yaml /stackable/jmx/region-server.yaml
RUN ln -s /stackable/jmx/restserver.yaml /stackable/jmx/rest-server.yaml
```

Comment (sbernauer) [reply hbase/Dockerfile:164]: Good catch, thx!

**Comment by siegfriedweber** [comment]
## Images

* docker.stackable.tech/stackable/hbase:2.4.12-stackable23.7.0-20231111
  Digest: sha256:5fd8644d54e4a6b36a9dcf6be6024244ae23ca1199e8a5b3d510163a453a633e
* docker.stackable.tech/stackable/hbase:2.4.17-stackable23.7.0-20231111
  Digest: sha256:14aa65db2492b3173c106b6ed80045569788cc59747d9b8b6c8553523c7ba3bf

built from branch [build-hbase-fix](https://github.com/stackabletech/docker-images/tree/build-hbase-fix), commit 7a1fcc994067eafcb854886d4e8d448d5142ca28

**Comment by siegfriedweber** [comment]
# Integration tests

Test branch: [hbase-operator#use-image-stackable23.7.0-20231111](https://github.com/stackabletech/hbase-operator/tree/use-image-stackable23.7.0-20231111), commit [16fb06c](https://github.com/stackabletech/hbase-operator/commit/16fb06c80f089c51b35f3e5690214e3de7c60587)

Cluster: OpenShift 4.12
Operators: commons=23.7, secret=23.7, zookeeper=23.7, hdfs=23.7, hbase=23.7
Product images:
* hbase
  * docker.stackable.tech/stackable/hbase:2.4.12-stackable23.7.0-20231111
  * docker.stackable.tech/stackable/hbase:2.4.17-stackable23.7.0-20231111
* hdfs
  * docker.stackable.tech/stackable/hdfs:3.3.4-stackable23.7.0-dns-fix
* zookeeper
  * docker.stackable.tech/stackable/zookeeper:3.3.4-stackable23.7.0-dns-fix

Result:
```
--- PASS: kuttl (1835.79s)
    --- PASS: kuttl/harness (0.00s)
        --- PASS: kuttl/harness/cluster-operation_hbase-latest-2.4.12-stackable23.7.0-20231111_hdfs-latest-3.3.4-stackable23.7.0-dns-fix_zookeeper-latest-3.8.1-stackable23.7.0-dns-fix (309.43s)
        --- PASS: kuttl/harness/smoke_hbase-2.4.12-stackable23.7.0-20231111_hdfs-3.3.4-stackable23.7.0-dns-fix_zookeeper-3.8.1-stackable23.7.0-dns-fix_listener-class-cluster-internal (315.49s)
        --- PASS: kuttl/harness/logging_hbase-2.4.17-stackable23.7.0-20231111_hdfs-latest-3.3.4-stackable23.7.0-dns-fix_zookeeper-latest-3.8.1-stackable23.7.0-dns-fix (338.10s)
        --- PASS: kuttl/harness/orphaned_resources_hbase-latest-2.4.17-stackable23.7.0-20231111_hdfs-latest-3.3.4-stackable23.7.0-dns-fix_zookeeper-latest-3.8.1-stackable23.7.0-dns-fix (337.67s)
        --- PASS: kuttl/harness/orphaned_resources_hbase-latest-2.4.12-stackable23.7.0-20231111_hdfs-latest-3.3.4-stackable23.7.0-dns-fix_zookeeper-latest-3.8.1-stackable23.7.0-dns-fix (260.85s)
        --- PASS: kuttl/harness/smoke_hbase-2.4.17-stackable23.7.0-20231111_hdfs-3.3.4-stackable23.7.0-dns-fix_zookeeper-3.8.1-stackable23.7.0-dns-fix_listener-class-external-unstable (325.62s)
        --- PASS: kuttl/harness/resources_hbase-latest-2.4.17-stackable23.7.0-20231111_hdfs-latest-3.3.4-stackable23.7.0-dns-fix_zookeeper-latest-3.8.1-stackable23.7.0-dns-fix (258.69s)
        --- PASS: kuttl/harness/resources_hbase-latest-2.4.12-stackable23.7.0-20231111_hdfs-latest-3.3.4-stackable23.7.0-dns-fix_zookeeper-latest-3.8.1-stackable23.7.0-dns-fix (252.79s)
        --- PASS: kuttl/harness/logging_hbase-2.4.12-stackable23.7.0-20231111_hdfs-latest-3.3.4-stackable23.7.0-dns-fix_zookeeper-latest-3.8.1-stackable23.7.0-dns-fix (270.01s)
        --- PASS: kuttl/harness/smoke_hbase-2.4.17-stackable23.7.0-20231111_hdfs-3.3.4-stackable23.7.0-dns-fix_zookeeper-3.8.1-stackable23.7.0-dns-fix_listener-class-cluster-internal (306.96s)
        --- PASS: kuttl/harness/smoke_hbase-2.4.12-stackable23.7.0-20231111_hdfs-3.3.4-stackable23.7.0-dns-fix_zookeeper-3.8.1-stackable23.7.0-dns-fix_listener-class-external-unstable (307.08s)
        --- PASS: kuttl/harness/cluster-operation_hbase-latest-2.4.17-stackable23.7.0-20231111_hdfs-latest-3.3.4-stackable23.7.0-dns-fix_zookeeper-latest-3.8.1-stackable23.7.0-dns-fix (287.59s)
PASS
```

**Comment by sbernauer** [comment]
@siegfriedweber please feel free to merge once you think it makes sense

**Comment by lfrancke** [comment]
Moved to done but not merged

**Comment by lfrancke** [comment]
I think it's just an approval that's missing. As I'm not sure if that was intentional I'm not going to do so now.

**Review by sbernauer — APPROVED** [review]
LGTM. Feel free to merge if you are confident enough :)

**Pull request merged**